### DISCUSSION TITLE
Blocks on layer 0 affect blocks on layer 3

### DISCUSSION THREAD

**Discussion by Trickiy** [Wohlstand/TheXTech#799] [Q&A]
### Describe the issue

Using the following code: #0 DeccelerateLayerX,3,0,0,0,0,0.02 BlockTrigger,0,1,2,1000,0,0 BlockTrigger,0,1,4,1001,0,0 #1000 AccelerateLayerX,3,-2,0,0,2,-0.1 #1001 AccelerateLayerX,3,2,0,0,2,0.1 #END , When I walk into the left or right side of a wood block that is on the default layer, while a layer 3 wood block is on screen it will get pushed and go faster than whats coded.

### A type of bug

Compatibility bug

### Does this issue happens always or randomly?

Always happens

### Version

1.3.7-dev

### Version Hash

#e25f296

### Platform

3DS

### Processor architecture

None

**Comment by ds-sloth** [comment]
First, LunaScript is deprecated, and should not be used for any new content. It may possibly be removed (or enabled only on our desktop builds) in the future. We plan to introduce Lua scripting in a future version, and that will be much more powerful and flexible.

But if you must create new content with LunaScript, then I can give you my advice. This seems like the intended behavior of the LunaScript commands you described. `BlockTrigger` is not meant to apply based on layers, but instead based on block type. Please let me know what happens if you make all of the blocks on Layer 3 a unique type, and then use that type with your BlockTrigger command.

**Comment by Trickiy** [comment]
I can't figure out how to make a block type unique but I can change what block I use. The one I used is a ice block on layer 3 and apon doing so the ice reacts in a more buggy way. When I walk into the ice it's nudged, but when I walk into the wood the ice slides continually. And if Mario is on top during he slides off the ice going the opposite direction at the same speed.

**Comment by ds-sloth** [comment]
This is a challenging thing to implement. Keep trying if you like, but I would suggest you'll find more success implementing similar functionality using NPC attach layers.

**Comment by ds-sloth** [comment]
Thanks for the report, I think this is more of a scripting help question than an engine bug. You're welcome to convert it to a discussion if you want more advice.

**Comment by Wohlstand** [comment]
> convert it to a discussion

That maintainers can do only. Actually this is really needs to be turned into discussion. I'll do that.
